Усманов БИВТ-24-14 Лабораторная №5

ByakuByaku opened the pull request:
24 в списке

NemirovichMark committed:
Add files via upload

NeAlyssa committed:
Merge pull request #1 from NemirovichMark/main

Update Lab5

NemirovichMark committed:
Update ProgramTests.cs test 1_1

NemirovichMark committed:
Update ProgramTests.cs test 4

NemirovichMark committed:
Update ProgramTests.cs

NemirovichMark committed:
Update Program.cs task 4

NeAlyssa committed:
Merge pull request #2 from NemirovichMark/main

Update Lab5

NemirovichMark committed:
Update ProgramTests.cs

NemirovichMark committed:
Update Program.cs comments 1_1 - 3_7 upd

NeAlyssa committed:
Merge pull request #4 from NemirovichMark/main

Upd Lab5

NemirovichMark committed:
Add files via upload

NemirovichMark committed:
Update Program.cs

NeAlyssa committed:
Merge pull request #5 from NemirovichMark/main

Upd Lab5

ByakuByaku committed:
Update Program.cs

NeAlyssa commented:
Добрый день. До субботы персонально Вашу работу я уже не проверю. Поэтому для получения допуска Вам нужно самостоятельно найти и исправить замечания, если они относятся к Вашей работе. Общая инструкция (самые частые ошибки, с которыми возвращаются 90% работ):

В 5 лабораторной работе необходимо стараться следовать принципу разработки ПО DRY (Don't Repeat Yourself). Более конкретно - избегать дублирование одинакового кода там, где это возможно.

В 2_2 вынесите одинаковый код в отдельный метод.
В 2_8 вынесите одинаковый код в отдельный метод.
В 2_20 вынесите одинаковый код в отдельный метод.
В 2_28b вынесите одинаковый код в отдельный метод.
В 2_28c вынесите одинаковый код в отдельный метод.

В 2_3 вынесите одинаковый код в отдельный метод.
В 2_27 вынесите одинаковый код в отдельный метод.

Не надо реализовывать методы для решения задания внутри самого метода с заданием. Из-за этого вспомогательные методы не могут быть использованы для решения других заданий без потенциального дублирования кода. Вытащите реализованные методы из методов с заданиями, удалите дублирующиеся.

Методы, которые реализуются в лабораторной, должны быть нестатическими. Статикой не стоит злоупотреблять.

И, конечно же, убедитесь, что Ваша работа проходит тесты. Чаще всего с непройденными тестами присылаются номера 3_5 (тесты необходимо обновить до последней версии), и 1_1 (отсутствует обработка краевых случаев).

ByakuByaku committed:
Update Program.cs

использовал DRY, подправил ошибки, прошелся по вашему списку уточнений, сократил код, все тесты работают

ByakuByaku commented:
Здравствуйте, проверил свой код на ваши уточнения, сократил код, убрав повторяющиеся элементы в методы, подправил 1.1, все тесты проходит

NeAlyssa commented:
Принято. Не забудьте загрузить работу в задание на Moodle для выставления баллов за допуск.
